### PR TITLE
✨ Helper script for a podman kind cluster with an image registry

### PR DESCRIPTION
**Pull request by itroyano**
# Description

Adding a helper script for a podman-based kind cluster with an image registry.
This should help local testing by speeding up initial setup, required for running Tilt.

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

**Comment by bentito** [comment]
Maybe it's time to create a `dev` dir or put this `hack`? Just I think the main dir is getting pretty crowded with single files.

**Comment by itroyano** [comment]
Agree @bentito.  I'll also add a md file in there now.

**Comment by itroyano** [comment]
@bentito added a couple changes -
1. dev folder
2. the gist as a readme md

File: dev/podman/setup-local-env-podman.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+### Verify installed tools (install if needed)
+
+```sh
+podman --version
+podman version 5.0.1
+kind version
+kind v0.23.0 go1.22.3 darwin/arm64
+tilt version
+v0.33.12, built 2024-03-28
```
Comment (tmshort) [inline dev/podman/setup-local-env-podman.md:11]: Minor nit: I like shell prompts in front of commands (e.g. `$`) to distinguish it from output.

File: dev/podman/setup-local-env-podman.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+### Start Kind with a local registry
+Use this [helper script](./kind-with-registry-podman.sh) to create a local single-node Kind cluster with an attached local image registry.
+
+You'll need to disable secure access on the local kind registry:
+
+`podman inspect kind-registry --format '{{.NetworkSettings.Ports}}'`
+With the port you find for 127.0.0.1 edit the config file:
```
Comment (tmshort) [inline dev/podman/setup-local-env-podman.md:20]: This seems incomplete.

Comment (tmshort) [reply dev/podman/setup-local-env-podman.md:20]: Is this supposed to be referencing the next section? In that case I'd suggest using 4 hashes for the section header below.

File: dev/podman/setup-local-env-podman.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+### Bind the local registry to the Podman machine
```
Comment (tmshort) [inline dev/podman/setup-local-env-podman.md:22]:
```suggestion
#### Bind the local registry to the Podman machine
```

Comment (tmshort) [reply dev/podman/setup-local-env-podman.md:22]: But only if this is meant to be a continuation of "edit the config file:"

File: dev/podman/setup-local-env-podman.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+### export DOCKER_HOST
+
+`export DOCKER_HOST=unix:///var/run/docker.sock`
+
+
+### Start tilt with the tilt file in the parent directory
+
+`DOCKER_BUILDKIT=0 tilt up`
+
```
Comment (tmshort) [inline dev/podman/setup-local-env-podman.md:44]: Should this be considered optional also? (Not everyone is using `tilt`, I haven't even set it up yet.)

File: dev/podman/kind-with-registry-podman.sh [hunk 5]
```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+set -o errexit
+
```
Comment (tmshort) [inline dev/podman/kind-with-registry-podman.sh:3]: Any need/desire to add a check here to make sure `podman` is actually installed? (e.g. test `command -v podman`)

File: dev/podman/kind-with-registry-podman.sh [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+# 1. Create registry container unless it already exists
+reg_name='kind-registry'
+reg_port='5001'
+if [ "$(podman inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)" != 'true' ]; then
+  podman run \
+    -d --restart=always -p "127.0.0.1:${reg_port}:5000" --network bridge --name "${reg_name}" \
```
Comment (tmshort) [inline dev/podman/kind-with-registry-podman.sh:9]: Should the `5000` port here (and below) also be configurable (or at least specified via variable)?

**Review by tmshort — COMMENTED** [review]
Only one "real" thing for the readme.

File: dev/podman/setup-local-env-podman.md (hunk 1 above)
Comment (itroyano) [reply dev/podman/setup-local-env-podman.md:11]: Missed that. Thanks!

File: dev/podman/setup-local-env-podman.md (hunk 2 above)
Comment (itroyano) [reply dev/podman/setup-local-env-podman.md:20]: This sections is off... I'll rephrase it to make sense.

File: dev/podman/setup-local-env-podman.md (hunk 4 above)
Comment (itroyano) [reply dev/podman/setup-local-env-podman.md:44]: Done

File: dev/podman/kind-with-registry-podman.sh (hunk 5 above)
Comment (itroyano) [reply dev/podman/kind-with-registry-podman.sh:3]: No need IMO since the readme mentions Podman as a prerequisite.

**Review by tmshort — APPROVED** [review]
/lgtm

**Review by joelanford — APPROVED** [review]
